libr/anal/p/anal_x86_cs.c

```diff
@@ -1013,8 +1013,17 @@ static void anop_esil (RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int len
 		{
 			char *src = getarg (&gop, 1, 0, NULL);
 			char *dst = getarg (&gop, 0, 1, "^");
-			esilprintf (op, "%s,%s,$z,zf,=,$p,pf,=,$s,sf,=,$0,cf,=,$0,of,=",
-				src, dst);
+			char *_dst;
+			if ((a->bits == 64) && dst && (strlen (dst) == 6) && (dst[0] == 'e') && (_dst = strdup (dst))) {
```

Comment:
Is strdup necessary for a 6 char string?

> On 24 Jun 2016, at 14:09, condret <notifications@github.com> wrote:
> 
> In libr/anal/p/anal_x86_cs.c:
> 
> > @@ -1013,8 +1013,17 @@ static void anop_esil (RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int len
> >  		{
> >  			char *src = getarg (&gop, 1, 0, NULL);
> >  			char *dst = getarg (&gop, 0, 1, "^");
> > -			esilprintf (op, "%s,%s,$z,zf,=,$p,pf,=,$s,sf,=,$0,cf,=,$0,of,=",
> > -				src, dst);
> > +			char *_dst;
> > +			if ((a->bits == 64) && dst && (strlen (dst) == 6) && (dst[0] == 'e') && (_dst = strdup (dst))) {
> so what do you want me to do about strlen + strdup?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>